stable_baselines3/common/env_checker.py

```diff
@@ -205,15 +206,33 @@ def _check_obs(obs: Union[tuple, dict, np.ndarray, int], observation_space: spac
     correspond to the declared one.
     """
     if not isinstance(observation_space, spaces.Tuple):
-        assert not isinstance(
-            obs, tuple
-        ), f"The observation returned by the `{method_name}()` method should be a single value, not a tuple"
+        # Accept a single-value tuple for GraphInstance for compatibility with other tuple types.
```

Comment:
I would not add more complexity here, just check if the observation space is a Graph and return early with a warning saying that Graph spaces are not supported and won't be checked further by the env checker.

Reply:
sure so I've defined the check for the type before the tuple type checks.